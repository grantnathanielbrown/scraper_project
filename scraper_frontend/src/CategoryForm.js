import React, { Component } from 'react'

export default class CategoryForm extends Component {
    constructor(props) {
        super(props)
        this.state= {
        }  

    }
             
    render() {
        return (
            <div className="category-form">
        <label>
          Category 	&nbsp;	&nbsp;	&nbsp;	&nbsp;	&nbsp;       
          <select id="category" value={this.state.categoryValue} onChange={this.props.handleChange}>
            <option value="Best">Best</option>
            <option value="Controversial">Controversial</option>
            <option value="Hot">Hot</option>
            <option value="New">New</option>
            {/* <option value="Random">Random</option> */}
            <option value="Rising">Rising</option>
            <option value="Top">Top</option>
          </select>
        </label>                 
            </div>
        )
    }
}
