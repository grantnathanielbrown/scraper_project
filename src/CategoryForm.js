import React, { Component } from 'react'

export default class CategoryForm extends Component {

    render() {
        return (
            <div className="category-form">
                <label>
                Category 	&nbsp;	&nbsp;	&nbsp;	&nbsp;	&nbsp;       
                <select value={this.props.selectedCategory} id="selectedCategory" onChange={this.props.handleChange}>                    
                    {/* <option value="Best">{this.props.initialCategory}</option> */}
                    <option value="Best">Best</option>
                    <option value="Controversial">Controversial</option>
                    <option value="Hot">Hot</option>
                    <option value="New">New</option>
                    {/* <option value="Random">Random</option> */}
                    <option value="Rising">Rising</option>
                    <option value="Top">Top</option>
                </select>
                </label>                 
            </div>
        )
    }
}
