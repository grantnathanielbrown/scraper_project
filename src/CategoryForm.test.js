import React from 'react';
import { shallow } from 'enzyme';
import {createSerializer} from 'enzyme-to-json';
import CategoryForm from './CategoryForm';


expect.addSnapshotSerializer(createSerializer({mode: 'deep'}));

describe('Term Form', () => {
    const mockHandleChange = jest.fn();    
    const categoryForm = shallow(<CategoryForm handleChange={mockHandleChange} />);
    it('renders properly', () => {
        expect(categoryForm).toMatchSnapshot();
    });
    it('can call the function that is passed down as a prop', () => {
        const selectedCategory = 'Controversial';
        categoryForm.find('#category').simulate('change', {target: {value: selectedCategory}});
        expect(mockHandleChange).toHaveBeenCalledTimes(1);
    });

    
});