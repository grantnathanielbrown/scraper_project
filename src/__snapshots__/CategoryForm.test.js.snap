// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Term Form renders properly 1`] = `
<div
  className="category-form"
>
  <label>
    Category           
    <select
      id="category"
      onChange={[MockFunction]}
    >
      <option
        value="Best"
      >
        Best
      </option>
      <option
        value="Controversial"
      >
        Controversial
      </option>
      <option
        value="Hot"
      >
        Hot
      </option>
      <option
        value="New"
      >
        New
      </option>
      <option
        value="Rising"
      >
        Rising
      </option>
      <option
        value="Top"
      >
        Top
      </option>
    </select>
  </label>
</div>
`;
